internal/controller/replication.storage/volumegroupreplication_controller.go

```diff
@@ -424,6 +430,10 @@ func (r *VolumeGroupReplicationReconciler) SetupWithManager(mgr ctrl.Manager) er
 
 			// Check if the pvc labels match any VGRs based on selectors present in it's annotation
 			for _, vgr := range vgrObjsList.Items {
+				// Skip reconcile for PVC that belongs to an external VGR
+				if vgr.Spec.External {
```

Comment:
For now this is good but even we need more enhancements for example not even cache this object when the External field is set

Reply:
https://github.com/csi-addons/kubernetes-csi-addons/issues/815 pointer it here so that the main issue contains a link to this PR where we need to address cache problem

Reply:
For the time being I have added the check in PVC predicate, it is beneficial in all the cases.